Java_Assignment-2/src/Product.java

```diff
@@ -0,0 +1,154 @@
+		while(choice=='y'){
+			cartItems.add(productList.get(sc.nextInt()));
+			System.out.println("Quantity: ");
+			quantity=sc.nextInt();
+			productQuantity.add(productQuantityIndex++, quantity);
+			
+			System.out.println("Do you want to add more items ?(y/n)");
+			choice=sc.next().charAt(0);
+		}
+		
+		showCartItems();
+		showChoiceMenu();
+	}
+	
+	private void removeFromCart(){
+		char choice='y';
+		while(choice=='y'){
+			System.out.println("Enter the item no. to remove it: ");
+			int removeIndex=sc.nextInt();
+			cartItems.remove(removeIndex-1);
+			System.out.println("Do you want to remove more items ?(y/n)");
+			choice=sc.next().charAt(0);
+			}
+		showCartItems();
+		showChoiceMenu();
+		}
+	
+	private void updateProductQuantity(){
+		System.out.println("Enter the product no.: ");
+		int productNumber=sc.nextInt();
+		System.out.println("updated quantity: ");
+		int productQuantity=sc.nextInt();
+		this.productQuantity.remove(productNumber-1);
+		this.productQuantity.add((productNumber-1), productQuantity);
+		showCartItems();
+		showChoiceMenu();
+	}
+	
+	private void showCartItems(){
+		System.out.println("Items in cart:");
+		int index=0;
+		for(Product items:cartItems){
+			System.out.println(items.productName+"\t"+productQuantity.get(index++)+"\t"+items.productPrice);
+		}
+	}
+	
+	public void showProducts(){
+		System.out.println("PRODUCTS AVAILABLE\n");
+		System.out.println("number"+"\t\t"+"Products"+"\t"+"MRP"+"\t\t"+"discount");
+		for(Product items:productList.values()){
+			System.out.println(items.productId+"\t\t"+items.productName+"\t\t"+items.productPrice+"\t\t"+items.discount+" %");
+		}
+	}
+	
+	private void showChoiceMenu(){
+		System.out.println("Enter your choice\n1.Add item to cart  2.remove any item  3.update quantity  4. proceed to bill");
+		int Choice=sc.nextInt();
+		switch(Choice){
+			case 1:
+				addToCart();
+				break;
+			case 2:
+				removeFromCart();
+				break;
+			case 3:
+				updateProductQuantity();
+				break;
+			case 4:
+				generateBill();
+				break;
+			default:
+				System.out.println("Incorrect choice...");
+		}
+	}
+	
+	private void generateBill(){
+		double totalAmount=0;
+		int index=0;
+		System.out.println("HAVING PROMOCODE ???(y/n)");
+		char choice=sc.next().charAt(0);
+		if(choice=='n'){
+		for(Product item:cartItems){
+			totalAmount+=(item.productPrice*productQuantity.get(index++));
+		 }
+		 System.out.println("Total Price = "+totalAmount+" /- only\nThanks for shopping.");
+		}
+		else{
+			System.out.println("Enter the Promocode: ");
+			String promocode=sc.next();
+			for(Product item:cartItems){
+				totalAmount+=(item.productPrice*productQuantity.get(index++));
+			 }
+				FixedOrderPromotion obj=new FixedOrderPromotion();
+				obj.setMinimumPrice(10000);
+				obj.setFixedDiscount(10);
+				if(obj.isPromotionApplicable(promocode, totalAmount)){
+				 System.out.println("Amount after discount: "+obj.getFixedDiscount(totalAmount));
```

Comment:
getFixedDiscount() method should return only discount amount the discounted bill amount.